media/js/src/CollectionTab.jsx

```diff
@@ -111,20 +111,20 @@ export default class CollectionTab extends React.Component {
                 );
             });
         } else {
-            // Sync the selected asset with the state of the
-            // collection's assets, in case the user added, edited, or
-            // removed any selections on this asset.
-            const newAssets = updateAsset(
-                this.props.assets,
-                // Filter the annotations of this selection based on
-                // the current asset filter state.
-                filterSelections(
-                    this.state.selectedAsset,
-                    this.state.title, this.state.owner,
-                    this.state.tags, this.state.terms,
-                    this.state.date
-                ));
-            this.props.onUpdateAssets(newAssets);
+            // Sync assets and tag/term state with the server.
```

Comment:
This is working *very well*, and does not seem slow. The filter values (owner, title, etc) are correctly retained as I toggle back and forth between full collection and item details. There is a bug with page numbers. Looking at "All Class Members" in course 84, there are 4 pages of items. Clicking into an item then "Back to Collection" results in the pagination controls displaying `1 of 1`.